wallet-next/src/storage.rs

```diff
@@ -1,30 +1,84 @@
+use std::path::PathBuf;
+
 use penumbra_chain::params::ChainParams;
 use penumbra_crypto::{
     merkle::{NoteCommitmentTree, Tree},
     note::Commitment,
     FieldExt,
 };
 use penumbra_proto::{crypto::FullViewingKey, Message, Protobuf};
-use sqlx::{query, Pool, Sqlite};
+use sqlx::{migrate::MigrateDatabase, query, Pool, Sqlite};
 
 use crate::sync::ScanResult;
 
+const MAX_MERKLE_CHECKPOINTS_CLIENT: usize = 10;
+
 #[derive(Clone)]
 pub struct Storage {
-    pool: Pool<Sqlite>,
+    storage_path: PathBuf,
 }
 
 impl Storage {
-    pub fn new(pool: Pool<Sqlite>) -> Self {
-        Self { pool }
+    pub fn load(storage_path: PathBuf) -> Self {
+        Self { storage_path }
     }
 
-    pub async fn migrate(self: &Storage) -> anyhow::Result<()> {
-        sqlx::migrate!().run(&self.pool).await.map_err(Into::into)
+    pub async fn initialize(
+        self: &Storage,
+        storage_path: PathBuf,
+        fvk: FullViewingKey,
+        params: ChainParams,
+    ) -> anyhow::Result<()> {
+        //   Check that the file at the given path does not exist;
+        if !storage_path.exists() {
+            // Create the SQLite database
+
+            let path_str = storage_path.as_os_str().to_str().unwrap();
```

Comment:
Is there a more succinct/less unwieldy method of converting `PathBuf` to `&str`?

Reply:
No, the `PathBuf` type is unwieldy because it isn't required to be UTF-8 (because system file paths aren't). But if SQLite wants a `&str`, then we could avoid the whole thing by replacing `PathBuf` with `String`, and then `&storage_path` could just be passed to `create_database` directly.